evm/governance.js

```diff
@@ -231,6 +257,67 @@ async function processCommand(options, chain) {
             break;
         }
 
+        case 'gatewayUpgrade': {
+            if (contractName === 'AxelarServiceGovernance') {
+                throw new Error(`Invalid governance action for AxelarServiceGovernance: ${governanceAction}`);
+            }
+
+            if (unixEta < getCurrentTimeInSeconds() + contractConfig.minimumTimeDelay && !yes) {
+                printWarn(`${eta} is less than the minimum eta.`);
+                const answer = readlineSync.question(`Proceed with ${governanceAction}?`);
+                if (answer !== 'y') return;
+            }
+
+            if (!isAddress(implementation) || implementation === AddressZero) {
+                throw new Error(`Invalid new gateway implementation address: ${implementation}`);
+            }
+
+            const gatewayContract = new Contract(target, IGateway.abi, wallet);
+            const targetInterface = new Interface(gatewayContract.interface.fragments);
+
+            const implementationCode = await provider.getCode(implementation);
+
+            if (implementationCode === '0x') {
+                printWarn(`There is no code deployed at ${implementation}`);
+                const answer = readlineSync.question(`Proceed with ${governanceAction}?`);
+                if (answer !== 'y') return;
+            }
+
+            const newGatewayImplementationCodeHash = keccak256(implementationCode);
+
+            const setupParams = '0x';
+            const upgradeCalldata = targetInterface.encodeFunctionData('upgrade', [
+                implementation,
+                newGatewayImplementationCodeHash,
+                setupParams,
+            ]);
+
+            values[2] = upgradeCalldata;
+
+            gmpPayload = defaultAbiCoder.encode(types, values);
```

Comment:
check if a non zero eta already exists for this proposal, and print a warning if so